### DISCUSSION TITLE
How get SteamID C#

### DISCUSSION THREAD

**Discussion by ghost** [SteamRE/SteamKit#916] [General]
Hello, i know the following questions is stupid for experimented programmers but i'm newbie in this...

I'm trying to get the SteamID like this:
string SteamID = SteamUser.LoggedOnCallback.ClientSteamID.AccountID;

what i'm doing wrong? Im using WinForms because i need collect information about steam for display in a form...

I appreciate all replys with constructive intentions!
Thanks for this tool guys :)

**Comment by yaakov-h** [comment]
What format of SteamID are you trying to land up with? They can be represented in many ways.

Check out the `Render` function(s?) on our SteamID class.

**Comment by ghost** [comment]
I dont understand, can you explain me?
I'm trying to get SteamID64 and Nickname of the user

![2020-09-08 (2)](https://user-images.githubusercontent.com/59315516/92511021-1aef9300-f1e3-11ea-8b5b-57bd2c077749.png)

**Reply by M0binMoharrami** [reply]
steamid is a ulong format and you have to convert that to string then you have to put it in a string type!
youre code should be like this
string SteamID = Convet.ToString(SteamUser.LoggedOnCallback.ClientSteamID.Render());

**Comment by oxters168** [comment]
LoggedOnCallback is an event to be subscribed to. You can have a function that gets called when the event fires. You can find an example of how to use it [here](https://github.com/SteamRE/SteamKit/blob/master/Samples/1.Logon/Program.cs). There are other useful examples [here](https://github.com/SteamRE/SteamKit/tree/master/Samples). Good luck

**Comment by ghost** [comment]
Oh now i understand the event. But i only need get a SteamID64 and username not the credentials... i think is more simple but i can't find the way...

**Comment by oxters168** [comment]
If you're trying to get the steamid from the username, then that's not possible ([reference](https://github.com/SteamRE/SteamKit/issues/279)). You'll need to log in as the user whose steamid you want. Once they're logged in, the LoggedOnCallback gets fired, and in that function you can get the steamid ([reference](https://github.com/SteamRE/SteamKit/issues/291)). You can also get the friends of the user and their shown names and steamids after logging in.

**Comment by ghost** [comment]
I think you dont understand me, im trying to get username and SteamID64 from client. For example, steam saves this on C:\Program Files (x86)\Steam\config\loginusers.vdf But i only want parse the steamid64 from there... Thanks for your replies

**Reply by guillerp8** [reply]
Use Gameloop Vdf to do that.

**Comment by oxters168** [comment]
Oh, I don't know if SteamKit is capable of doing that. From my understanding SteamKit "emulates" the steam client. I don't think it has a way of reading the actual steam client's loginusers.vdf file. I could be mistaken, hopefully someone else here can help answer that.
